src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/common/BerConverter.cs

```diff
@@ -549,11 +549,11 @@ private static int EncodingMultiByteArrayHelper(SafeBerHandle berElement, byte[]
                         IntPtr valPtr = Marshal.AllocHGlobal(structSize);
                         Marshal.StructureToPtr(managedBervalArray[i], valPtr, false);
 
-                        tempPtr = checked((IntPtr)((long)berValArray + IntPtr.Size * i));
+                        tempPtr = berValArray + IntPtr.Size * i;
```

Comment:
This package multi-targets download runtimes where this will use checked arithmetic, so this is actually introducing a bug. Could you please convert the method to use proper unmanaged pointers.

Reply:
(same comment for other files in System.DirectoryServices.Protocols

Reply:
Hm, not sure what is the proper unmanaged pointer should  be, all APIs used here require pass/return IntPtr ...

Reply:
I would use `void*` or `byte*` as much as possible, and cast these when interfacing with these legacy APIs.